### DISCUSSION TITLE
Add an option to modify the style of the base of a flat ride

### DISCUSSION THREAD

**Discussion by frutiemax** [OpenRCT2/OpenRCT2#15764] [Feature requests]
### Describe the feature

![image](https://user-images.githubusercontent.com/12629817/131423008-785e2258-7a44-4f27-98e8-8988f12f9795.png)

Add an option to modify the style of the base of the ride in the Color Scheme Options tab as shown in the above picture. 

### Feature type

Quality of life

**Comment by spacek531** [comment]
I'm told this is NSF material

**Comment by spacek531** [comment]
I'd like it if we could choose any available path type as the base in addition to invisible. It's a tiny bit of RCT3 functionality that I think I wouldn't mind having.

**Comment by frutiemax** [comment]
This is definitely NSF as you need new flags for the base appearance.

**Comment by spacek531** [comment]
I mean hidden base is specifically on the radar of NSF devs

**Comment by frutiemax** [comment]
I hardcoded it for fun.

**Comment by spacek531** [comment]
![image](https://user-images.githubusercontent.com/3454156/131438327-66e70165-851b-4235-ba1d-84d9fe1500b1.png)
NSF right now

This isn't exactly what you describe, and I think there's functionality missing from merely having a station object decide.

I'd go with 3 states in a new field:
- default
- invisible floor and fences
- path (the path that connects to the exit)

The path option could alternatively be a list of available path types.

The path option would go hand-in-hand with a new entrance hut reminiscent of those in RCT3 that copies its ground texture from the connected path.

**Comment by Gymnasiast** [comment]
You can implement all of it in `develop`, except for the part where the user can actually change the selection. Station styles, for example, are already using objects and you can actually change the selection if you unhide the tab (don’t try to save your game if you do, it’s hidden for that reason).

If we want to have this selectable, I think it might be better to introduce objects for "station platform style". There are already two styles: standard (wooden platforms for tracked rides, brown base for flat) and invisible (under development). Yours would then be a third one.

**Comment by spacek531** [comment]
I had the same idea that station platforms could be changed with objects. I didn't decide if a new object type would be necessary or if the right flags in "station" objects would suffice.

I'm not sure about extrapolating the default wood platforms. There's a lot of behavior specific to different ride types:
- each inverted coaster gets its own set of station supports
- there's 4 types of metal station bases which are used on different rides
- there's different size stations for different rides
- boat hire dock

The station buildings can be housed in such an object just fine.

For a platform object, I think these behaviors and sprites would be necessary:
- Flag to indicate swinging ride behavior
- Flag to indicate flat ride behavior
- Flag to indicate invisibility
- Flag for 1 base sprite (existing behavior)
- Flag for 4 base sprites (existing behavior)
- Flag for 21 base sprites (path border behavior)
- Flag to indicate base has glass sprites
- Flag to indicate base sprites are directional (like Spiral Slide base)
- Flag to indicate a roof is present for dodgems
- Flag to indicate roof has glass sprites

**Comment by karst** [comment]
This is somewhat added now with NSF, it's part of the station style though, so not seperate. It could be neat to be able to choose between transparent, dirt, padded ground, tiles and connected exit path though.
